src/logging/WriterFrontend.cc

```diff
@@ -133,6 +133,16 @@ WriterFrontend::~WriterFrontend() {
 }
 
 void WriterFrontend::Stop() {
+    if ( disabled ) {
+        for ( int j = 0; j < write_buffer_pos; j++ )
+            DeleteVals(num_fields, write_buffer[j]);
+
+        delete[] write_buffer;
+        write_buffer = nullptr;
+        write_buffer_pos = 0;
+        return;
+    }
+
     FlushWriteBuffer();
```

Comment:
Since this calls `FlushWriteBuffer()` anyway, you could just drop that change here — unless you wanted it in both methods for robustness (in which case you could make it a private method).

Reply:
It's more about bailing out of `Stop` early if it's disabled. I hoisted it into a private method. I could call `FlushWriteBuffer` and then check for `disabled`, but it'd mean that `Stop` is relying on a side-effect of `FlushWriteBuffer` for the cleanup which isn't ideal.